src/portable/macos.rs

```diff
@@ -288,6 +289,12 @@ pub fn stop_and_disable(name: &str) -> anyhow::Result<bool> {
         log::info!("Removing unit file {}", unit_path.display());
         fs::remove_file(unit_path)?;
     }
+
+    // Clear the runstate dir - macOS wouldn't delete UNIX domain socket
+    // files after server shutdown, which may lead to issues in upgrades
+    fs::remove_dir_all(runstate_dir(name)?).ok();
```

Comment:
I think we should filter and remove only domain sockets if that is the problem. Something like this?

```
    for entry in fs::read_dir(dir_path)? {
        let entry = entry?;
        let path = entry.path();

        if let Ok(metadata) = path.metadata() {
            if metadata.file_type().is_socket() {
                fs::remove_file(&path)?;
            }
        }
    }
```

Reply:
Yup, that'll do! Do you mind pushing that to the PR? I'm kinda in the middle of home moving. Thank you!